src/test/routes/safe/store/actions/utils.test.ts

```diff
@@ -0,0 +1,170 @@
+import { getLastTx, getNewTxNonce, shouldExecuteTransaction } from 'src/logic/safe/store/actions/utils'
+import { getMockedSafeInstance, getMockedTxServiceModel } from 'src/test/utils/safeHelper'
+import axios from 'axios'
+import { buildTxServiceUrl } from 'src/logic/safe/transactions'
+
+describe('shouldExecuteTransaction', () => {
+  it('It should return true if given a safe with a threshold > 1',  async () => {
+    // given
+    const nonce = '0'
+    const threshold = '2'
+    const safeInstance = getMockedSafeInstance({ threshold })
+    const lastTx = getMockedTxServiceModel({})
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(false)
+  })
+  it('It should return true if given a safe with a threshold === 1 and the last transaction is first transaction (nonce 0)',  async () => {
+    // given
+    const nonce = '0'
+    const threshold = '1'
+    const safeInstance = getMockedSafeInstance({ threshold })
+    const lastTx = getMockedTxServiceModel({})
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return true if given a safe with a threshold === 1 and the last transaction is not the first transaction, the last transaction is already executed',  async () => {
+    // given
+    const nonce = '10'
+    const threshold = '1'
+    const safeInstance = getMockedSafeInstance({ threshold })
+    const lastTx = getMockedTxServiceModel({ isExecuted: true })
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(true)
+  })
+  it('It should return false if given a safe with a threshold === 1 and the last transaction is not first transaction, the last transaction is not already executed',  async () => {
+    // given
+    const nonce = '10'
+    const threshold = '1'
+    const safeInstance = getMockedSafeInstance({ threshold })
+    const lastTx = getMockedTxServiceModel({ isExecuted: false })
+
+    // when
+    const result = await shouldExecuteTransaction(safeInstance, nonce, lastTx)
+
+    // then
+    expect(result).toBe(false)
+  })
+})
+
+describe('getNewTxNonce', () => {
+  it('It should return 2 if given the last transaction with nonce 1',  async () => {
+    // given
+    const safeInstance = getMockedSafeInstance({})
+    const lastTx = getMockedTxServiceModel({ nonce: 1 })
+    const expectedResult = '2'
+
+    // when
+    const result = await getNewTxNonce(undefined, lastTx, safeInstance)
+
+    // then
+    expect(result).toBe(expectedResult)
+  })
+  it('It should return 0 if given a safe with the last nonce 0 and no lastTransaction, also should call safeInstance nonce method',  async () => {
```

Comment:
`It should return 0 if given a safe with nonce 0 and no transactions should use safe contract instance for retrieving nonce`

I wouldn't include implementation details in the description